### PR TITLE
[terra-modal manager] Implemented focustrap to confine focus within the modal popup.

### PR DESCRIPTION
**Pull request by AH106586Harika**
### Summary
Fixed- Focus is continuing to go outside of the Modal Manager when navigating through the component. 

**What was changed:**
Added FocusTrap to confine the focus within the modal popup

**Why it was changed:**

Pressing the TAB key causes the focus to shift away from the modal popup.

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

[UXPLATFORM-10090](https://jira2.cerner.com/browse/UXPLATFORM-10090)

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-abstract-modal/src/_ModalContent.jsx [hunk 1]
```diff
@@ -141,61 +143,65 @@ const ModalContent = forwardRef((props, ref) => {
...
-                  useText = text.join('');
-                }
-                return (
-                  <VisuallyHiddenText text={useText} />
-                );
-              }}
-            </FormattedMessage>
-          )}
+      <FocusTrap>
+        <div>
```
Comment (supreethmr) [inline packages/terra-abstract-modal/src/_ModalContent.jsx:153]: I think div is not required here. any specific reason to have an wrapper div. 
FocusTrap needs focusable child elements, `div` would violate this rule of focustrap

File: packages/terra-framework-docs/src/terra-dev-site/test/modal-manager/DisclosureIntegrationComponent.jsx [hunk 2]
```diff
@@ -139,7 +139,7 @@ class DisclosureComponent extends React.Component {
           {this.renderFormButton()}
           <br />
           <br />
-          <SelectField label="T-shirt size" placeholder="Select a size" selectId="tshirt-size-field" required className={cx('select-field')}>
+          <SelectField variant="combobox" label="T-shirt size" placeholder="Select a size" selectId="tshirt-size-field" required className={cx('select-field')}>
```
Comment (supreethmr) [inline packages/terra-framework-docs/src/terra-dev-site/test/modal-manager/DisclosureIntegrationComponent.jsx:142]: @AH106586Harika have you created separate defect to fix the issue of terra-form-single-select not working within modal ..??

**Comment by AH106586Harika** [comment]
> @AH106586Harika Focus seems to be lost on notification-dialog open previously focus was on button as shown in this screen shots https://github.com/cerner/terra-framework/pull/1964/files#diff-4fcb341919dfbd704aa64d99068e8e098f65764c1766b8c95b49b419a8aa82ad

Updated

File: packages/terra-framework-docs/src/terra-dev-site/test/modal-manager/DisclosureIntegrationComponent.jsx (hunk 2 above)
Comment (AH106586Harika) [reply packages/terra-framework-docs/src/terra-dev-site/test/modal-manager/DisclosureIntegrationComponent.jsx:142]: Raised Defect for form-single-select: [UXPLATFORM-10169](https://jira2.cerner.com/browse/UXPLATFORM-10169)

**Review by saket2403 — APPROVED** [review]

**Review by supreethmr — CHANGES_REQUESTED** [review]
For passivity concerns focustrap should be optional and can be enabled by having `shouldTrapFocus` prop. Since react-focus-trap has issue with portaled elements we need to look for other options to confine the focus within modal content.

**Comment by AH106586Harika** [comment]
The use of focus trap for the single select dropdown within the modal manager popup was unsuccessful, leading to the decision to abandon the PR.

**Pull request closed without merging**